### PR TITLE
Update QIDO to respect QueryStatus and header to include Erroneous attribute

### PR DESCRIPTION
**Pull request by pengchen0692**
## Description
Update QIDO to respect QueryStatus and header to include Erroneous attribute

## Related issues
Addresses [AB#84774](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/84774)

## Testing
All tests pass

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Core/DicomCoreResource.resx [hunk 1]
```diff
@@ -414,4 +414,8 @@ The first part date {1} should be lesser than or equal to the second part date {
   <data name="InvalidIncludeAllFields" xml:space="preserve">
     <value>Invalid QIDO-RS query. Cannot specify included fields in addition to 'all'.</value>
   </data>
+  <data name="QueryIsDisabledOnTag" xml:space="preserve">
+    <value>Query is disabled on specified query tag '{0}'.</value>
```
Comment (pengchen0692) [inline src/Microsoft.Health.Dicom.Core/DicomCoreResource.resx:418]: Change to specified attribute to keep consistency.

File: src/Microsoft.Health.Dicom.Core/Features/Query/Model/QueryExpression.cs [hunk 2]
```diff
@@ -66,6 +69,8 @@ public class QueryExpression
         /// </summary>
         public IReadOnlyCollection<QueryFilterCondition> FilterConditions { get; }
 
+        public IReadOnlyCollection<string> ErroneousTags { get; }
```
Comment (pengchen0692) [inline src/Microsoft.Health.Dicom.Core/Features/Query/Model/QueryExpression.cs:72]: add summary

File: src/Microsoft.Health.Dicom.Api/Extensions/HttpResponseExtensions.cs [hunk 3]
```diff
@@ -12,12 +13,27 @@ namespace Microsoft.Health.Dicom.Api.Extensions
 {
     internal static class HttpResponseExtensions
     {
+        public const string ErroneousAttributesHeader = "x-ms-erroneous-attributes";
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Api/Extensions/HttpResponseExtensions.cs:16]: Let's not use `x-ms`. This convention has been deprecated. Perhaps just `erroneous-attributes` or `erroneous-dicom-attributes`

**Review by wsugarman — APPROVED** [review]

File: src/Microsoft.Health.Dicom.Api/Extensions/HttpResponseExtensions.cs (hunk 3 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Api/Extensions/HttpResponseExtensions.cs:16]: As the header conversation scope is 'global', I feel we may want to indicate `dicom`. will use `erroneous-dicom-attributes` .

**Pull request merged**